### PR TITLE
close and reset Python subprocess state on clear

### LINKED ISSUE #1015: close Python subprocess on `clear all` or `clear functions`
Use `clear all` or `clear functions` after creating some symbolic variables, and the persistent function state is reset, but the pipe and subprocess are left open.

First reported on the Octave bug tracker [here](https://savannah.gnu.org/bugs/?58070).

### PR DESCRIPTION
**Pull request by mtmiller**
Move popen2 reset logic to a new helper function, and call it using a
persistent onCleanup object.

Fixes #1015.

### CODE REVIEW

File: inst/private/python_ipc_popen2.m [hunk 1]
```diff
@@ -43,24 +43,11 @@
...
-      pid = [];
-      A = (t == 0);
-    else
-      A = true;
-    end
-    if (~isempty(fout))
-      t = fclose(fout); fout = [];
-      A = A && (t == 0);
-    end
+    cleanup = [];
```
Comment (mtmiller) [inline inst/private/python_ipc_popen2.m:46]: This assignment triggers a call to the new reset helper function

File: inst/private/python_ipc_popen2.m [hunk 2]
```diff
@@ -43,24 +43,11 @@
...
+    fin = [];
+    fout = [];
+    pid = [];
+    A = true;
```
Comment (mtmiller) [inline inst/private/python_ipc_popen2.m:50]: Unfortunately I can't figure out a way to both use `onCleanup` and capture the return value

File: inst/private/python_ipc_popen2_reset.m [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @deftypefun {@var{A} =} python_ipc_popen2_reset (@var{fin}, @var{fout}, @var{pid})
+%% Private helper function for Python IPC.
+%%
+%% @var{A} is the resulting object, which might be an error code.
+%% @end deftypefun
+
+function A = python_ipc_popen2_reset (fin, fout, pid)
```
Comment (mtmiller) [inline inst/private/python_ipc_popen2_reset.m:26]: Maybe no point in returning any status here now?

File: inst/private/python_ipc_popen2_reset.m [hunk 4]
```diff
@@ -0,0 +1,51 @@
+%% Copyright (C) 2020 Mike Miller
```
Comment (mtmiller) [inline inst/private/python_ipc_popen2_reset.m:1]: This file was copied from `python_ipc_popen2.m`, do you think me should propagate the original Copyright here?

File: inst/private/python_ipc_popen2_reset.m (hunk 3 above)
Comment (cbm755) [reply inst/private/python_ipc_popen2_reset.m:26]: I think I agree.  I'm merging this to a local branch so I can try it on CI as well as try removing these return vlaues.

**Pull request merged**